valohai_cli/commands/deployment/create_version/dynamic_creation_command.py

```diff
@@ -0,0 +1,121 @@
+from valohai_cli.utils import humanize_identifier, parse_environment_variable_strings, sanitize_option_name
+
+DATUM_URI_PREFIX = 'datum://'
+
+
+class CreationCommand(click.Command):
+    """
+    A dynamically-generated subcommand that has Click options for files for deployments.
+    """
+    def __init__(
+        self,
+        project: Project,
+        deployment: Dict[str, Any],
+        endpoints: List[str],
+        commit: str,
+        version_name: str,
+        inherit_env_vars: bool = True,
+        environment_variables: Optional[Dict[str, str]] = None,
+    ):
+        self.project = project
+        self.deployment = deployment
+        self.endpoints = endpoints
+        self.commit = commit
+        self.version_name = version_name
+        self.environment_variables = dict(environment_variables or {})
+        self.inherit_env_vars = inherit_env_vars
+        super().__init__(
+            name=sanitize_option_name(deployment['name'].lower()),
+            callback=self.execute,
+            epilog='Multiple files per input: --my-input=myurl --my-input=myotherurl',
+            add_help_option=True,
+        )
+
+        commit_config = request(
+            method='get',
+            url=f'/api/v0/commits/{commit}/',
+            params={'include': 'config'},
+        ).json()
+        self.config = valohai_yaml.parse(commit_config['config'])
+
+        for endpoint in self.config.endpoints.values():
+            if endpoint.name in self.endpoints:
+                for file in endpoint.files:
+                    self.params.append(self.convert_file_to_option(endpoint_name=endpoint.name, file=file))
+
+    def convert_file_to_option(self, endpoint_name: str, file: File) -> Option:
+        """
+        Convert an endpoint file into a click Option.
+        """
+        name = f"{endpoint_name}.{file.name}"
+        option = click.Option(
+            param_decls=list(generate_sanitized_options(name)),
+            required=True,
+            metavar='URL',
+            help=f'File "{humanize_identifier(name)}"',
+        )
+        option.name = f"^{name}"
+        option.help_group = 'File Options'  # type: ignore[attr-defined]
+        return option
+
+    def execute(self, **kwargs: Any) -> None:
+        """
+        Execute the creation of the deployment version.
+
+        This is the Click callback for this command.
+
+        :param kwargs: Assorted kwargs (as passed in by Click).
+        :return: Naught
+        """
+        payload = {
+            'commit': self.commit,
+            'deployment': self.deployment['id'],
+            'environment_variables': parse_environment_variable_strings(self.environment_variables),
+            'inherit_environment_variables': self.inherit_env_vars,
+            'name': self.version_name,
+            'enabled': True,
+        }
+
+        endpoint_configurations: Dict[str, Any] = {}
+        endpoint_names_from_config = [e.name for e in self.config.endpoints.values()]
```

Comment:
A set would be more correct here, but a minor nit really.
```suggestion
        endpoint_names_from_config = {e.name for e in self.config.endpoints.values()}
```